test/cql-pytest/test_keyspace.py

```diff
@@ -76,6 +76,14 @@ def test_create_keyspace_nonexistent_dc(cql):
         ks = unique_name()
         cql.execute(f"CREATE KEYSPACE {ks} WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'nonexistentdc' : 1 }}")
 
+# Reproduces #20097
+def test_create_table_in_nonexistent_keyspace(cql):
+    table = unique_name()
+    with pytest.raises(InvalidRequest, match="nonexistent_keyspace"):
+        cql.execute(f"CREATE TABLE nonexistent_keyspace.{table} (p int PRIMARY KEY)")
+        # won't be reached, but just in case the CREATE wrongly succeeds
+        cql.execute(f"DROP TABLE nonexistent_keyspace.{table}")
```

Comment:
I just wonder this is supposed to only do cleanup in some rare cases but it can acctually hide a bug when create table succeeds (i.e. without actually creating) but drop somehow throws invalid request.

Reply:
You're right. Even better is to wrap the execute of DROP TABLE in a try that ignores all exceptions.

Reply:
I think that's an extreme case, because the KS existance is validated many times when CREATE TABLE is being processed, so even if I comment out my check, the statement is still rejected because the KS doesn't exist, which is checked later (again and again). Because of this low probability, I was reluctant to make this change, as the same reasoning, where a couple of testcase instructions hide a bug, can be applied to other testcases probably, but still I made it, though a bit differently from what @nyh suggested.